dbg.h

```diff
@@ -79,6 +81,27 @@ inline bool isColorizedOutputEnabled() {
 }
 #endif
 
+namespace timestamp {
+
+// Provide a timestamp at the precision of microsecond.
+// Not exactly the current microsecond yet, still a useful reference.
```

Comment:
Well, one option would be to make `timestamp_t` an empty `struct` and only determine the time when it's actually printed.

Reply:
This option sounds so genius! I cannot agree anymore. From your suggestion I've got another idea, how about just using `struct time {};`?